api/apps/api/test/access-control/project-acl-get.e2e-spec.ts

```diff
@@ -0,0 +1,132 @@
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { getFixtures } from './project-acl.fixtures';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+afterEach(async () => {
+  await fixtures?.cleanup();
+});
+
+test(`getting project users when user is not added to project`, async () => {
+  const projectId = await fixtures.GivenProjectWasCreated();
+  const response = await fixtures.WhenGettingProjectUserAsNotInProject(
+    projectId,
+  );
+  fixtures.ThenForbiddenIsReturned(response);
+});
+
+test(`getting project users as the owner and unique user`, async () => {
+  const projectId = await fixtures.GivenProjectWasCreated();
+  const response = await fixtures.WhenGettingProjectUsersAsOwner(projectId);
+  fixtures.ThenSingleOwnerUserInProjectIsReturned(response);
+});
+
+test(`getting project users with user in project but is viewer`, async () => {
+  const projectId = await fixtures.GivenProjectWasCreated();
+  await fixtures.WhenAddingANewViewerToTheProjectAsOwner(projectId);
```

Comment:
as Kamil pointed out below, this should not be an action but another condition - a name change alone may make the setup clearer:

```suggestion
  await fixtures.GivenAUserWasGivenViewerRoleOnProject(projectId);
```
Also, we shouldn't care to encode in the condition description that this was done by an owner, I think - if this were not the case, the role granting operation should be rejected, and we test that already. Incidentally, the actions in the tests for this (only owners can grant roles), you have correctly used `When` and encoded that the fact that the operation is carried out by a user with a given role.